### DISCUSSION TITLE
Creating a new NATS connection per each subscriber

### DISCUSSION THREAD

**Discussion by vedadiyan** [nats-io/nats-server#3941] [General]
I am working on creating a fully automated microservice framework and I found that if I create a new connection for each subscriber I may get the following benefits:

- I can name the connection and I can get out-of-the-box health monitoring through the built-in monitoring endpoint
- I can simplify service binding by eliminating global variables. This is useful if two microservices would need to connect to two different servers, however this is a very rare case. 

However, my concern is: 

- In case there would be too many subscribers and in case each subscriber runs in cluster mode by being attached to a queue, would it cause port exhaustion in the server? 
- Will the server get slow due to too many connection? 

In my opinion this approach has it's pros and cons, but I need to clarify this so I really appreciate if someone helps me better analyze this case to make sure it is production wise.

**Comment by derekcollison** [comment]
We would generally not suggest a connect per subscriber. However sometimes this becomes a natural pattern if an app has one main subscription.

When you ask the system about connections via the nats cli, you can filter those connections based on a subject. So you could monitor all connections that have a subscription to say `foo.bar`.

**Reply by vedadiyan** [reply]
Thanks
